### PR TITLE
💬 Dedicated localization file + french language fixes.

### LINKED ISSUE #42: Strings into a separated file
Take all the strings of the app, and put them together in a `Localizable.strings` file.

Then create an enum (or struct with constants) to use them.

```swift
struct LocalizedString {

  static let example = "example".localized()
  // ...

}
```
(This is an example where `example` is the string id in the `Localizable.strings` file)

### LINKED ISSUE #43: Replace `Centre` occurrences by `Lieu`
After #42, we can then easily replace occurrences of `Centre` by `Lieu` in the `Localizable.strings` file.

Linked to https://github.com/CovidTrackerFr/vitemadose-front/issues/18 (follow what was done on frontend and android)

### LINKED ISSUE #57: App language is set to English on the App Store
Not sure if it is the right place to tell you this but FYI the app language is set to English on the App Store product page. 

Capture d’écran 2021-04-22 à 15 26 02

### PR DESCRIPTION
**Pull request by PlugNPush**
Fix #42 & Fix #43 & Fix #57

### CODE REVIEW

**Review by NathanFallet — CHANGES_REQUESTED** [review]
Duplicate: The String extension already exists, please delete the extension you added (String+Commons.swift)

**Comment by PlugNPush** [comment]
> Duplicate: The String extension already exists, please delete the extension you added (String+Commons.swift)

Objection your honor: it was not in the Xcode project, I looked for it for about 10 minutes scanning each file individually looking for the String Extension. 😅

**Comment by NathanFallet** [comment]
Now the question, as I mentioned in #42, where is the constant enumeration struct?
Let's see if @victor-sarda is okay to merge without it, or if he requests changes.

**Review by NathanFallet — DISMISSED** [review]

**Comment by victor-sarda** [comment]
Since the issue was assigned to me, I started working on it locally. I'm happy to approve after some changes in the String keys. I'll let you know 👍 
@NathanFallet an enum would be nice indeed! If it's too much work we can merge without and add it in another PR 😊

**Comment by PlugNPush** [comment]
Woi (working on it)

File: ViteMaDose/Views/CountySelection/Header/CountySelectionHeaderView.swift
Comment (NathanFallet) [inline ViteMaDose/Views/CountySelection/Header/CountySelectionHeaderView.swift:15]: Useless constant, you can use `LocalizedString.select_your_area` and `LocalizedString.area` directly instead of `titleText` and `highlightedText`.

File: ViteMaDose/Views/Home/Cells/HomeTitleCell.swift
Comment (NathanFallet) [inline ViteMaDose/Views/Home/Cells/HomeTitleCell.swift:66]: Same here

File: ViteMaDose/Views/Home/Cells/HomeTitleCell.swift
Comment (NathanFallet) [inline ViteMaDose/Views/Home/Cells/HomeTitleCell.swift:90]: Same here

File: ViteMaDose/Views/Home/Footer/HomePartnersFooterView.swift
Comment (NathanFallet) [inline ViteMaDose/Views/Home/Footer/HomePartnersFooterView.swift:20]: Here too

**Review by NathanFallet — CHANGES_REQUESTED** [review]
Some constant can be replaced by `LocalizedString.id`.

File: ViteMaDose/Views/Home/Footer/HomePartnersFooterView.swift
Comment (PlugNPush) [reply ViteMaDose/Views/Home/Footer/HomePartnersFooterView.swift:20]: Got it!

**Review by NathanFallet — DISMISSED** [review]

**Comment by PlugNPush** [comment]
This PR should now be safe for production :)

**Comment by PlugNPush** [comment]
Merge WIP

**Comment by PlugNPush** [comment]
@NathanFallet @victor-sarda Ready for a new review :)

**Comment by PlugNPush** [comment]
DO NOT ACCEPT THE PR, SWIFTLINT DOES NOT APPROVE IT.

**Comment by PlugNPush** [comment]
Should now be OK for production, due to the amount of code changed please review carefully. It is SwiftLint compliant and seems to run well in the simulator. Builds with no error (except the pre-existing deprecation warning).

**Comment by PlugNPush** [comment]
Since this PR got quite messy, should I close it and open a new one with a single all-in-one commit in it?

**Comment by victor-sarda** [comment]
> Since this PR got quite messy, should I close it and open a new one with a single all-in-one commit in it?

Can you close it and push a branch to the project with your changes please? You can do one single commit and I'll pick your stuff in my branch and open a PR on my side with clean commits 👍

**Comment by PlugNPush** [comment]
Cleaning things up!

**Pull request closed without merging**
